Terraform/Home_Task_2/modules/asg/main.tf

```diff
@@ -13,18 +13,19 @@ data "aws_ami" "amazon-linux-2" {
 
 }
 
+data "aws_ssm_parameter" "key_pair" {
+  name = "TEST"
+}
+
 resource "aws_launch_configuration" "asg_lc" {
-  name_prefix       = "terraform_lc"
-  image_id          = data.aws_ami.amazon-linux-2.id
-  instance_type     = "t2.micro"
-  key_name          = "TEST"
-  user_data         = file("user_data.sh")
-  placement_tenancy = "default"
-  security_groups   = [var.sg_public_1_id]
+  name_prefix     = "terraform_lc"
+  image_id        = data.aws_ami.amazon-linux-2.id
+  instance_type   = "t2.micro"
+  key_name        = data.aws_ssm_parameter.key_pair.name
+  user_data       = file("${path.module}/user_data.sh")

```

Comment:
It's ok with path.module, but you should also move your user_data.sh script to folder /modules/asg/**files**/user_data.sh. I've written about this before!